docs/tutorials/azureml.md

```diff
@@ -0,0 +1,335 @@
+        'token_type_ids':  torch.ones([1, 32], dtype=torch.long)     # dummy list of ones
+    }
+
+symbolic_names = {0: 'batch_size', 1: 'max_seq_len'}
+torch.onnx.export(model,                                         # model being run
+                  (inputs['input_ids'],
+                   inputs['attention_mask'], 
+                   inputs['token_type_ids']),                    # model input (or a tuple for multiple inputs)
+                  model_path,                                    # where to save the model (can be a file or file-like object)
+                  opset_version=11,                              # the ONNX version to export the model to
+                  do_constant_folding=True,                      # whether to execute constant folding for optimization
+                  input_names=['input_ids',
+                               'input_mask', 
+                               'segment_ids'],                   # the model's input names
+                  output_names=['start_logits', "end_logits"],   # the model's output names
+                  dynamic_axes={'input_ids': symbolic_names,
+                                'input_mask' : symbolic_names,
+                                'segment_ids' : symbolic_names,
+                                'start_logits' : symbolic_names, 
+                                'end_logits': symbolic_names})   # variable length axes
+```
+
+### Run the ONNX model with ONNX Runtime
+
+The following code runs the ONNX model with ONNX Runtime. You can test it locally before deploying it to Azure Machine Learning.
+
+The `init()` function is called at startup, performing the one-off operations such as creating the tokenizer and the ONNX Runtime session.
+
+The `run()` function is called when we run the model using the Azure ML endpoint. Add necessary preprocess() and postprocess() steps.
+
+For local testing and comparison, you can also run the PyTorch model.
+
+```python
+import os
+import logging
+import json
+import numpy as np
+import onnxruntime
+import transformers
+import torch
+
+# The pre process function take a question and a context, and generates the tensor inputs to the model:
+# - input_ids: the words in the question encoded as integers
+# - attention_mask: not used in this model
+# - token_type_ids: a list of 0s and 1s that distinguish between the words of the question and the words of the context
+# This function also returns the words contained in the question and the context, so that the answer can be decoded into a phrase. 
+def preprocess(question, context):
+    encoded_input = tokenizer(question, context)
+    tokens = tokenizer.convert_ids_to_tokens(encoded_input.input_ids)
+    return (encoded_input.input_ids, encoded_input.attention_mask, encoded_input.token_type_ids, tokens)
+
+# The post process function takes the list of tokens in the question and context, as well as the output of the 
```

Comment:
The post-process function takes the list of tokens in the question and context, the output of the model, the list of log probabilities for the choices of start and end of the answer, and maps it back to an answer to the question that is asked of the context.

Reply:
Reworded